src/richie/apps/core/api.py

```diff
@@ -0,0 +1,45 @@
+"""
+API endpoints for the core app.
+"""
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
+from .serializers import UserSerializer
+
+
+class UsersViewSet(ViewSet):
+    """
+    API endpoints to access users.
+    """
+
+    # pylint: disable=no-self-use,unused-argument
+    def list(self, request, version):
+        """
+        List requests for users. We do not provide an API to get a list of registered users at
+        this time, and therefore return 501 Not Implemented.
+        """
+        return Response(status=501)
+
+    # pylint: disable=no-self-use,invalid-name,unused-argument
+    def retrieve(self, request, pk, version):
+        """
+        Single item requests for users. We do not provide an API to get registered user by id
+        at this time, and therefore return 501 Not Implemented.
+        """
+        return Response(status=501)
+
+    @action(detail=False)
+    # pylint: disable=no-self-use,invalid-name,unused-argument
+    def whoami(self, request, version):
```

Comment:
from django.contrib.auth.decorators import login_required
from django.utils.decorators import method_decorator 

```suggestion
    @method_decorator(login_required)
    def whoami(self, request, version):
```

Reply:
Does this not redirect to a login page? (this is what I understand form the docs https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-login-required-decorator)

Reply:
I'm not sure but your test should tell. The correct way to do it otherwise should be to raise the `NotAuthenticated` exception: https://www.django-rest-framework.org/api-guide/exceptions/#notauthenticated

Reply:
- the `@method_decorator(login_required)` does a `302` redirect which is not what we want on an API endpoint

- `raise NotAuthenticated` returns a `403` which seems less appropriate than a `401` as this route **is intended to be used by unauthenticated users**